scripts/exregional_process_bufr.sh

```diff
@@ -0,0 +1,415 @@
+#
+print_info_msg "
+========================================================================
+Entering script:  \"${scrfunc_fn}\"
+In directory:     \"${scrfunc_dir}\"
+
+This is the ex-script for the task that runs bufr (cloud, metar, lightning) preprocess
+with FV3 for the specified cycle.
+========================================================================"
+#
+#-----------------------------------------------------------------------
+#
+# Specify the set of valid argument names for this script/function.  
+# Then process the arguments provided to this script/function (which 
+# should consist of a set of name-value pairs of the form arg1="value1",
+# etc).
+#
+#-----------------------------------------------------------------------
+#
+valid_args=( "CYCLE_DIR" "WORKDIR")
+process_args valid_args "$@"
+#
+#-----------------------------------------------------------------------
+#
+# For debugging purposes, print out values of arguments passed to this
+# script.  Note that these will be printed out only if VERBOSE is set to
+# TRUE.
+#
+#-----------------------------------------------------------------------
+#
+print_input_args valid_args
+#
+#-----------------------------------------------------------------------
+#
+# Load modules.
+#
+#-----------------------------------------------------------------------
+#
+case $MACHINE in
+#
+"WCOSS_C" | "WCOSS")
+#
+
+  . /apps/lmod/lmod/init/sh
+  module purge
+  module use /scratch4/NCEPDEV/nems/noscrub/emc.nemspara/soft/modulefiles
+  module load intel/16.1.150 impi/5.1.1.109 netcdf/4.3.0 
+  module list
+  
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="mpirun -l -n 1"
+  ;;
+#
+"HERA")
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="srun"
+  ;;
+#
+"JET")
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="srun"
+  ;;
+#
+"ODIN")
+#
+  module list
+
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="srun"
+  ;;
+#
+esac
+#
+#-----------------------------------------------------------------------
+#
+# Extract from CDATE the starting year, month, day, and hour of the
+# forecast.  These are needed below for various operations.
+#
+#-----------------------------------------------------------------------
+#
+set -x
+START_DATE=$(echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/')
+YYYYMMDDHH=$(date +%Y%m%d%H -d "${START_DATE}")
+JJJ=$(date +%j -d "${START_DATE}")
+
+YYYY=${YYYYMMDDHH:0:4}
+MM=${YYYYMMDDHH:4:2}
+DD=${YYYYMMDDHH:6:2}
+HH=${YYYYMMDDHH:8:2}
+YYYYMMDD=${YYYYMMDDHH:0:8}
+
+YYJJJHH=$(date +"%y%j%H" -d "${START_DATE}")
+PREYYJJJHH=$(date +"%y%j%H" -d "${START_DATE} 1 hours ago")
+
+#
+#-----------------------------------------------------------------------
+#
+# Get into working directory
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "$VERBOSE" "
+Getting into working directory for BUFR obseration process ..."
+
+cd ${WORKDIR}
+
+fixdir=$FIX_GSI
+fixgriddir=$FIX_GSI/${PREDEF_GRID_NAME}
+
+print_info_msg "$VERBOSE" "fixdir is $fixdir"
```

Comment:
`fixdir` is only used once below. Consider using `$FIX_GSI` directly on line 171 and removing this renaming of a global variable.

Please also do this in the other scripts.

Reply:
removed this one too